File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-204513-77784/workdir/src/surveycrawler/crawler_displacement.py
from surveycrawler.crawler_movement import CrawlerMovement


class CrawlerDisplacement(CrawlerMovement):

    # initialization

    def __init__(self, a_displacement):
        self._displacement = a_displacement

    # undoing

    def undo(self, a_crawler):
        a_crawler.move_by(self._displacement.negated())

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-210827-1958/analysis/own-tests-project/src/main/java/surveycrawler/CrawlerDisplacement.java
package surveycrawler;

public class CrawlerDisplacement extends CrawlerMovement {

    private final Point displacement;

    // initialization

    public CrawlerDisplacement(Point aDisplacement) {
        displacement = aDisplacement;
    }

    // moving

    @Override
    public void applyTo(SurveyCrawler aCrawler) {
        aCrawler.applyDisplacement(displacement);
    }

    @Override
    public void undoOn(SurveyCrawler aCrawler) {
        aCrawler.applyDisplacement(displacement.negated());
    }
}
